cartography/intel/aws/ec2/launch_templates.py

```diff
@@ -176,11 +164,18 @@ def sync_ec2_launch_templates(
     for region in regions:
         logger.info(f"Syncing launch templates for region '{region}' in account '{current_aws_account_id}'.")
         templates = get_launch_templates(boto3_session, region)
-        versions, found_templates = get_launch_template_versions(boto3_session, region, templates)
+        versions = get_launch_template_versions(boto3_session, region, templates)
+
+        # Ensure we only load templates for which we found versions
+        # (handles templates deleted between the two API calls)
+        valid_template_ids = {v['LaunchTemplateId'] for v in versions}
+        existing_templates = [t for t in templates if t['LaunchTemplateId'] in valid_template_ids]
```

Comment:
Is it possible for this be moved into the `transform_launch_templates()` function? Ideally the sync function only does a get+transform+load+cleanup; but completely understand if that ends up breaking some interface or making something else much more complicated